### PR TITLE
test(POD-034): failure-injection mid-transcribe — pins NFR-5 / AC-US-6.3 (Tier 3)

### PR DESCRIPTION
**Pull request by invaderDMG**
## Summary
- New Tier 3 test `tests/integration/test_failure_injection.py` — spawns the CLI as a real subprocess, watches stderr for the locked `event=transcribe_start`, SIGTERMs the child mid-transcribe, asserts no Markdown landed and no `.tmp` debris survived.
- Pins NFR-5 + AC-US-6.3 end-to-end against the **signal-termination** path that no in-process unit test can reach.
- Closes **POD-034**, the last open SP-6 task. SP-6 (M-6 — test pyramid + `--debug`) goal is now reachable.

## Acceptance criteria satisfied
The SRS traceability matrix (`SRS.md:498`) names this test verbatim — *"NFR-5 → AC-US-6.3 → Failure-injection integration test (kill mid-transcribe; assert no output file)"*. Internal ACs:

- [x] **AC-1** — kill happens mid-transcribe (after observing `event=transcribe_start` on stderr); child exits non-zero. `proc.returncode != 0` assertion.
- [x] **AC-2** — resolved output `.md` does not exist after the kill (atomic-rename invariant, ADR-0005). `not output_path.exists()` assertion.
- [x] **AC-3** — output parent dir has no `*.tmp` debris. `list(tmp_path.glob("*.tmp")) == []` assertion.
- [x] **AC-4** — module marked `pytest.mark.slow`, lives at `tests/integration/test_failure_injection.py`, win32-skipped (POSIX-only), mirrors the fixture pattern from `test_tty_harness.py` (POD-035).

## Edge cases covered
- [x] **Vacuous-kill detection** — preconditions check (`saw_transcribe_start` + non-zero exit) ensures (2) and (3) aren't no-ops. A clean exit would mean SIGTERM arrived after `atomic_write` succeeded; the test fails loudly with a diagnostic about bumping `_STARTUP_TIMEOUT_S` if the fixture grew.
- [x] **Cold-cache CI runs** — `_STARTUP_TIMEOUT_S = 240.0` mirrors `test_tty_harness.py`'s budget; absorbs ~30 s of faster-whisper tiny model download on a fresh runner.
- [x] **Reader-thread deadlock** — stderr is drained on a background reader thread feeding a `queue.Queue`; main thread polls with timeout. No risk of `readline()` blocking forever on a partial line if the child stops emitting but doesn't close stderr.
- [x] **Resource cleanup** — `proc.stderr.close()` in the `finally` block prevents the GC'd `FileIO` from raising `ResourceWarning` (which `filterwarnings = ["error"]` would upgrade to a hard test failure). Hit this in development; commit message documents the trap.
- [x] **Child reaping** — `proc.kill()` + `proc.wait(timeout=5)` fallback in `finally` ensures no leaked zombie if the SIGTERM-then-wait cycle exceeds `_KILL_TIMEOUT_S` (30 s).

## Why SIGTERM, not SIGINT
ADR-0014 catches SIGINT at the cli surface and converts it to a clean exit-1 path that runs Python cleanup (`atexit` hooks, `finally` blocks). SIGTERM is uncaught by default — the process dies immediately. Under SIGTERM, **the only thing keeping the target path clean is the design**: no tempfile is ever created before `atomic_write` opens one near the very end of `Pipeline.run`. That's a stronger contract than SIGINT would test.

## Risks touched
- **NFR-5** (atomic output) — directly verified end-to-end. The unit-tier `tests/unit/test_atomic_write.py` covers in-process exception cleanup; this test covers the harder signal-kill path.
- **AC-US-6.3** (refuse-overwrite preserves prior file on transcribe failure) — the same atomic-rename contract; the negative-of-positive form (no output instead of preserved-prior) because we kill before any output existed.
- **Test pyramid** (M-6 sprint goal) — fills the last failure-injection gap noted in PROJECT_PLAN §6 (SP-6 cross-cutting list).

## Documentation updated
- [x] `CHANGELOG.md` `[Unreleased] ### Added` — bullet under "Test fixtures + quality gates" naming the contract pinned, the kill point, the SIGTERM rationale, and the three load-bearing assertions.

## Test plan
- [x] `uv run pytest tests/integration/test_failure_injection.py -m slow -v` → PASSED (6.24 s warm-cache; cold-cache adds ~30 s for tiny model download).
- [x] `uv run pytest -q` → 252 passed (Tier 1 unchanged; deselected count went from 26 → 27 reflecting the new slow-marked test).
- [x] `uv run pytest -m slow -q` → **6 passed** in 33 s (was 5; +1 = this test).
- [x] `uv run pytest -m contract -q` → 21 passed (Tier 2 unaffected).
- [x] `uv run ruff check` / `ruff format --check` / `mypy --strict` → clean (44 files; was 43).
- [ ] CI matrix (Ubuntu + macOS-14): expected green; both runners hit the slow-tier step per `.github/workflows/ci.yml:79-84`.

## Definition of Done
- [x] Trunk-based feature branch, squash-merge target.
- [x] Tier 3 test added; default suite ≤ 1 s preserved (1.39 s observed); Tier 3 step picks up the new test via `-m slow`.
- [x] CHANGELOG `[Unreleased]` updated.
- [x] Story status: POD-034 finished; SP-6 cross-cutting list closed.
- [ ] CI matrix box (filled post-merge).
- [ ] Maintainer signoff at sprint review (per `PROJECT_PLAN.md` §1.5 — story status tracked in GitHub Issues, sprint actuals to `CHANGELOG.md`).

Refs NFR-5, AC-US-6.3, ADR-0005, ADR-0012, ADR-0014, ADR-0017.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

**Review by invaderDMG — COMMENTED** [review]
## Review summary

**Verdict:** approve (posted as `COMMENT` because GitHub rejects self-APPROVE with HTTP 422; user granted explicit `--override-self-review`).

**Findings:**
- 0 blocking
- 0 should-fix
- 2 suggestions
- 3 praise

**Quality gate:** ran locally on `a2c86a7` —
- `uv run ruff check` clean
- `uv run ruff format --check` clean (44 files)
- `uv run mypy --strict src tests` clean
- `uv run pytest -q` → 252 passed, 27 deselected (1.34 s — default suite ≤ 1 s budget honoured)
- `uv run pytest -m contract -q` → 21 passed (8.68 s)
- `uv run pytest tests/integration/test_failure_injection.py -m slow -v` → 1 passed (5.81 s)
- CI: both `ubuntu-latest` and `macos-14` green per `gh pr view`. ✅

**Top-level notes:**
- **Scope:** clean. Diff is exactly the two files the PR body promises (`CHANGELOG.md` + `tests/integration/test_failure_injection.py`); POD-034 in `PROJECT_PLAN.md:223` matches.
- **AC coverage (NFR-5 / AC-US-6.3):** `SRS.md:498` traceability row names this test verbatim. The three contract assertions are guarded by two precondition assertions (`saw_transcribe_start`, `proc.returncode != 0`) that prevent the test from passing vacuously if the kill timing drifts — the right shape for an integration test of a design-level invariant.
- **Architecture:** correct Tier 3 placement per ADR-0017; the `*.tmp` glob assertion lines up with `atomic_write`'s `suffix=".md.tmp"` (atomic_write.py:39); SIGTERM-not-SIGINT rationale lines up with ADR-0014 (which catches SIGINT and converts to a clean exit-1 path).
- **Standards:** ruff / ruff-format / mypy --strict all clean on the new file (44 files; +1 from main).
- **Security:** zero new deps, no new external fetches, paths flow from `tmp_path` + in-repo fixture, subprocess via argv list (no `shell=True`).
- **Docs:** CHANGELOG `[Unreleased] ### Added` entry present, references PR #32 (backfilled from `#TBD` in commit `a2c86a7`), names contract / kill point / SIGTERM rationale / three load-bearing assertions.

**Praise (worth calling out):**
1. The precondition assertions are how integration tests *should* guard themselves — without them a test that drifted out of timing would silently start passing for the wrong reason.
2. The `proc.stderr.close()` in `finally` plus the comment explaining `filterwarnings = ["error"]` would upgrade `ResourceWarning` to a hard failure documents a concrete trap contributors would otherwise repeat.
3. The module docstring's three thought experiments ("moving `atomic_write` earlier would fail (3); writing direct to target would fail (2); kill-too-late would fail (1)") explain why the *set* of assertions pins the design, not the implementation. Future-me reading this in two years will know what to keep when refactoring.

**Suggestions (non-blocking):**
1. Line-level: drop or update the `pipeline.py:222` reference in the module docstring — the `event=transcribe_start` literal is at line 224 (`_log.info` call begins at line 221). Line numbers in docstrings drift; anchoring to the symbol is more durable. (See line-level suggestion block.)
2. **Out of scope for this PR — follow-up only:** `REPO_ROOT`, `SAMPLE_MP3`, `_require_ffmpeg`, `sample_mp3` are now duplicated across 4 Tier 3 files (test_integration_tiny.py, test_integration_ec3.py, test_tty_harness.py, test_failure_injection.py). `_cli_argv` across 2. Rule of Three crossed; a small `tests/integration/conftest.py` consolidation would tidy this — but it doesn't belong in POD-034.

Nothing is blocking. Quality gate green locally, CI green, every PR-body claim verified. Approving as `COMMENT` per the self-review constraint above.

---
_Self-review note: this review was conducted under explicit user override of the no-self-review rule (`/pr-review 32 --override-self-review`). If the verdict is wrong, dismiss with: `gh api -X PUT /repos/invaderDMG/podcast-script/pulls/32/reviews/<id>/dismissals -f message="reviewed in error"`._

File: tests/integration/test_failure_injection.py [hunk 1]
```diff
@@ -0,0 +1,245 @@
...
+cleanup (atexit hooks, finally blocks). SIGTERM is uncaught: the
+process dies immediately, no Python-level cleanup runs, and the only
+thing keeping the target path clean is the design (no tempfile is ever
+created before atomic_write opens one). That's a stronger test than
+SIGINT.
+
+The kill is timed against the locked ``event=transcribe_start``
+phase-boundary token (ADR-0012 / 22-token catalogue, emitted by
+:meth:`Pipeline._transcribe_speech` at line 222 of
+``src/podcast_script/pipeline.py``). We tail the child's stderr in a
```
Comment (invaderDMG) [inline tests/integration/test_failure_injection.py:42]: **[suggestion]** [docs] — line number `222` is drifty

The `event=transcribe_start` literal lives at `src/podcast_script/pipeline.py:224` today; the `_log.info(...)` call begins at line 221. Either is fine to anchor on, but `222` matches neither. Line numbers in docstrings tend to drift with the next refactor — anchoring to the symbol is more durable.

```suggestion
phase-boundary token (ADR-0012 / 22-token catalogue, emitted by
:meth:`Pipeline._transcribe_speech` in
``src/podcast_script/pipeline.py``). We tail the child's stderr in a
```

Non-blocking; feel free to ignore if you prefer to keep the line-number anchor and just bump it.
